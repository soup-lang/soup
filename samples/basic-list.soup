func write_0_2(x, y, z)
    console.log(x)
    console.log(z)

write_0_2([false, true], [0, 1], 3.1415926)

y: 3.14159
x: [y, 2.71828]
console.log(x)

console.log([] ++ [])
console.log([] ++ [0])
console.log([0] ++ [])
console.log([0] ++ [0])
z: [1.19]
console.log(z++x)

console.log([].length)
console.log([].length = 0)
ls0: []
ls0.push(0)
console.log(ls0)

console.log([0].length)
console.log([1].length = 0)
ls1: [2]
ls1.push(0)
console.log(ls1)

ls2: [z]
ls2.push([3.14159])
console.log(ls2)
