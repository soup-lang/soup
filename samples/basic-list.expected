[ false, true ]
3.1415926
[ 3.14159, 2.71828 ]
[]
[ 0 ]
[ 0 ]
[ 0, 0 ]
[ 1.19, 3.14159, 2.71828 ]
0
true
[ 0 ]
1
false
[ 2, 0 ]
[ [ 1.19 ], [ 3.14159 ] ]
